include/revng/PipelineC/Prototypes.h

```diff
@@ -275,6 +301,47 @@ rp_kind *rp_step_get_artifacts_kind(rp_step *step);
  */
 rp_container *rp_step_get_artifacts_container(rp_step *step);
 
+/**
+ * \return the number of analysis present in this step
+ */
+int rp_step_get_analysis_count(rp_step *step);
+
+/**
+ * \return the of the analysis in the provided step with the provided index.
+ *
+ * index must be less than rp_step_get_analysis_count
+ */
+rp_analysis *rp_step_get_analysis(rp_step *step, int index);
+
+/**
+ * \return the count of containers used by the provided analysis.
+ * is no analysis
+ */
+int rp_analysis_get_arguments_count(rp_analysis *analysis);
+
+/**
+ * \return a non owning pointer to the name of the container used as index
+ * argument of the analysis of this step.
+ *
+ * index must be less than rp_step_get_analysis_arguments_count(step)
+ */
+const char *rp_analysis_get_argument_name(rp_analysis *analysis, int index);
+
+/**
+ * \return the quantity of kinds that can be accepted by a analysis
+ */
+int rp_analysis_get_argument_acceptable_kinds_count(rp_analysis *analysis,
+                                                    int argument_index);
+
+/**
+ * \return the pointer to a acceptable kind for the container with index
+ * argument_index within a analysis, nullptr if kind_index is >= than
+ * rp_analysis_argument_acceptable_kinds_count(analysis, argument_index)
+ */
+const rp_kind *rp_analysis_get_argument_acceptable_kind(rp_analysis *analysis,
+                                                        int argument_index,
+                                                        int kind_index);
```

Comment:
Would dropping `_acceptable` take away any valuable information?

Reply:
it can always be put in the comment, but i feelt hat `get argument kind` is saying that the argument has a kind, `acceptable` makes it more clear it is a filter